components/heap/include/esp_heap_debug.h

```diff
@@ -0,0 +1,39 @@
+#ifndef _ESP_HEAP_DEBUG_H
+#define _ESP_HEAP_DEBUG_H
+
+#ifdef CONFIG_HEAP_TASK_TRACKING
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum region_types {
+    DRAM = 0,
+    D_IRAM = 1,
+    IRAM = 2,
+    SPIRAM = 3,
+    NUM_USED_TYPES = 4
+};	
+
+typedef struct {
+    TaskHandle_t task;
+    int size[NUM_USED_TYPES];
+} heap_dump_totals_t;
+
+typedef struct {
+    TaskHandle_t task;
+    void* address;
+    int size: 24;                   // The size of the allocated block.
+    int type: 7;                    // Type of block's region
```

Comment:
In keeping with the suggestions I made above, I'd prefer if this type included a `uint32_t caps` member rather than creating a new enum for expressing the memory type.

Reply:
Addressed in the reply above.

Reply:
Still pending further discussion.